### PR TITLE
Update Arbitrum's vendored contracts to Nitro

### PR DESCRIPTION
**Pull request by Amxx**
Vendor contract from [arbitrum nitro](https://github.com/OffchainLabs/nitro/tree/master/contracts/src)

Some types have been changed (for example from `IOwnable` to `address`. The change are marked by 
```// OZ: retyped from xxx```

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Changelog entry

### CODE REVIEW

**Comment by istankovic** [comment]
Thanks @Amxx for submitting this. :purple_heart:

File: contracts/vendor/arbitrum/IArbSys.sol [hunk 1]
```diff
@@ -1,63 +1,60 @@
-// SPDX-License-Identifier: MIT
-// OpenZeppelin Contracts (last updated v4.6.0) (vendor/arbitrum/IArbSys.sol)
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
```
Comment (frangio) [inline contracts/vendor/arbitrum/IArbSys.sol:2]:
```suggestion
// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,7 @@
  * `Ownable2Step`: extension of `Ownable` that makes the ownership transfers a two step process. ([#3620](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3620))
  * `Math` and `SignedMath`: optimize function `max` by using `>` instead of `>=`. ([#3679](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3679))
  * `Math`: Add `log2`, `log10` and `log256`. ([#3670](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3670))
+ * `Venor`: Update the vendored arbitrum contracts to match the nitro upgrade. ([#3692](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3692))
```
Comment (frangio) [inline CHANGELOG.md:35]:
```suggestion
 * Arbitrum: Update the vendored arbitrum contracts to match the nitro upgrade. ([#3692](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3692))
```

**Comment by frangio** [comment]
> Some types have been changed (for example from `IOwnable` to `address`. The change are marked by
> `// OZ: retyped from xxx`

Do we need this for some technical reason?

Is this PR ready to be merged?

**Comment by Amxx** [comment]
> > Some types have been changed (for example from `IOwnable` to `address`. The change are marked by
> > `// OZ: retyped from xxx`
> 
> Do we need these annotations for some technical reason? I would prefer to remove them.

It did not feel right to vendor a modified version of Arbitrum's code without documenting the changes.

**Pull request merged**